actioncable/lib/action_cable/engine.rb

```diff
@@ -22,7 +22,12 @@ class Engine < Rails::Engine # :nodoc:
 
     initializer "action_cable.set_configs" do |app|
       options = app.config.action_cable
-      options.allowed_request_origins ||= /https?:\/\/localhost:\d+/ if ::Rails.env.development?
+      if ::Rails.env.development? && options.allowed_request_origins.nil?
+        local_ips = ["localhost"] + Socket.ip_address_list.select{|ip| ip.ipv4?}.map do |ip|
+          ip.ip_address
+        end
+        options.allowed_request_origins = local_ips.map {|ip| Regexp.new("https?:\/\/#{ip}:\\d+")}
+      end
```

Comment:
After pondering on it for awhile I think a proc would be more useful in cases where we identify a range of clients, such as with config.web_console.whitelisted_ips.  (Or maybe even better there would be to simply directly support CIDR addresses so it's friendlier for network-centric people.)  For our current focus -- allowed_request_origins -- since they simply specify a protocol and the rails server endpoint, the current string / Regexp approach is pretty appropriate.

As a side-note, based on the name I bet that many people are initially confused thinking allowed_request_origins is a setting to specify client machines and not the Rails server(s), so there could be a better description of that in the docs.

I have thought more about enhancing security during development while making this as easy to develop with as possible, and have decided to only make all IPs available if someone binds 0.0.0.0.  Otherwise just open up the single IP that they specify.  (And still always include localhost.)